.github/workflows/website.yml

```diff
@@ -76,3 +77,68 @@ jobs:
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - uses: actions/cache@v2
+        continue-on-error: true
+        timeout-minutes: 2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-pkg-mod-${{ hashFiles('go.sum') }}
+      - run: |
+          # go get github.com/katbyte/terrafmt
+          git clone --branch json-output --single-branch https://github.com/gdavison/terrafmt terrafmt
+          cd terrafmt
+          go install
+      - run: curl -s https://raw.githubusercontent.com/terraform-linters/tflint/master/install_linux.sh | sh
+      - run: |
+          exit_code=0
+
+          # Configure the rules for tflint.
+          # The *_invalid_* rules disabled here prevent evaluation of expressions.
+          # Do not disable *_invalid_name rules, since these are good checks for e.g. "%s" formatting verbs
+          # being carried over from test cases.
+          shared_rules=(
+              "--enable-rule=terraform_comment_syntax"
+              "--disable-rule=aws_cloudwatch_event_target_invalid_arn"
+              "--disable-rule=aws_cognito_user_pool_domain_invalid_domain"
+              "--disable-rule=aws_db_instance_default_parameter_group"
+              "--disable-rule=aws_elasticache_cluster_default_parameter_group"
+              "--disable-rule=aws_iam_saml_provider_invalid_saml_metadata_document"
+              "--disable-rule=aws_iam_server_certificate_invalid_certificate_body"
+              "--disable-rule=aws_iam_server_certificate_invalid_private_key"
+              "--disable-rule=aws_transfer_ssh_key_invalid_body"
+              "--disable-rule=aws_worklink_website_certificate_authority_association_invalid_certificate"
+          )
+          find ./website/docs -type f \( -name '*.md' -o -name '*.markdown' \) \
+            | sort -u \
+            | while read -r filename ; do
+            rules=("${shared_rules[@]}")
+            if [[ "$filename" == "./website/docs/guides/version-2-upgrade.html.md" ]]; then
+                # ./website/docs/guides/version-2-upgrade.html.md should still include pre-0.12 syntax,
+                # since v1.0 does not support Terraform 0.12.
+                rules+=(
+                "--disable-rule=terraform_deprecated_interpolation"
+                "--disable-rule=terraform_deprecated_index"
+                )
+            else
+                rules+=(
+                "--enable-rule=terraform_deprecated_interpolation"
+                "--enable-rule=terraform_deprecated_index"
+                )
+            fi
+            # echo "Let's go with $filename..."
+            # We need to capture the output and error code here. We don't want to exit on the first error
+            set +e
+            ./scripts/validate-terraform-file.sh "$filename" "${rules[@]}"
+            lint_exitcode=$?
+            set -e
+            if [ $lint_exitcode -ne 0 ]; then exit_code=1; fi
+          done
+
+          exit $exit_code
```

Comment:
This could be moved to yet another script. It's messy because of the special handling for `./website/docs/guides/version-2-upgrade.html.md`. Otherwise, the `read` could be moved into the script, and it could be called like

```bash
find ./website/docs -type f \( -name '*.md' -o -name '*.markdown' \) | ./scripts/validate-terraform.sh "${rules[@]}"
```

That model could be used for the acceptance test validation